include/ck_tile/core/arch/arch.hpp

```diff
@@ -60,13 +61,253 @@ enum struct memory_operation_enum : std::uint16_t
     add
 };
 
-CK_TILE_HOST_DEVICE constexpr index_t get_warp_size()
+/*! @enum amdgcn_target_arch_id
+ * @brief Defines constants for AMDGCN architecture target IDs
+ */
+enum struct amdgcn_target_arch_id
 {
-#if defined(__GFX9__) || !defined(__HIP_DEVICE_COMPILE__)
-    return 64;
-#else
-    return 32;
-#endif
+    GFX908  = 0x0908,
+    GFX90A  = 0x090A,
+    GFX942  = 0x0942,
+    GFX950  = 0x0950,
+    GFX1100 = 0x1100,
+    GFX1101 = 0x1101,
+    GFX1102 = 0x1102,
+    GFX1151 = 0x1151,
+    GFX1200 = 0x1200,
+    GFX1201 = 0x1201,
+    HOST    = 0x0000,
+};
+
+/*! @enum amdgcn_wave_size
+ * @brief Defines constants for AMDGCN architecture wave sizes
+ */
+enum struct amdgcn_wave_size
+{
+    WAVE32 = 32u,
+    WAVE64 = 64u,
+    HOST   = 1u,
+};
+
+/**
+ * @brief Converts a lower-case string to the corresponding amdgcn_target_arch_id value.
+ *        Returns amdgcn_target_arch_id::HOST if no match is found.
+ *        Matches if the input contains the architecture substring.
+ *        Example: "gfx908", "gfx90a", "gfx1100", etc. can be parsed from hip runtime info.
+ */
+constexpr inline auto gfx_target_string_to_arch_id(char const* testStr)
```

Comment:
Can we be more specific about what the functions does. Why do we need find? If it's for strings like "gfx908;gfx942" do we care about the order of checks. Is this runtime or build time? Do we want an error path? Is HOST the right default.

Reply:
Yes this one is meant to parse the gfxtarget string from the hipDevice runtime. Might be better to put in common code for tests.